builder/api/src/main/java/io/helidon/builder/api/ProvidedUtil.java

```diff
@@ -0,0 +1,192 @@
+                     Class<T> configType,
+                     List<ConfiguredService> configuredServices,
+                     boolean allFromServiceLoader) {
+        Map<String, ConfiguredProvider<T>> allProvidersByType = new HashMap<>();
+        Map<String, ConfiguredProvider<T>> unusedProvidersByType = new LinkedHashMap<>();
+        serviceLoader.forEach(it -> {
+            allProvidersByType.put(it.configKey(), it);
+            unusedProvidersByType.put(it.configKey(), it);
+        });
+
+        List<T> result = new ArrayList<>();
+
+        // first add all configured
+        for (ConfiguredService service : configuredServices) {
+            ConfiguredProvider<T> provider = allProvidersByType.get(service.type());
+            if (provider == null) {
+                throw new ConfigException("Unknown provider configured. Expecting a provider with type \"" + service.type()
+                                                  + "\", but only the following providers are supported: "
+                                                  + allProvidersByType.keySet() + ", "
+                                                  + "provider interface: " + providerType.getName()
+                                                  + ", configured service: " + configType.getName());
+            }
+            unusedProvidersByType.remove(service.type());
+            if (service.enabled()) {
+                result.add(provider.create(service.serviceConfig(), service.name()));
+            }
+        }
+
+        // then (if desired) add the rest
+        if (allFromServiceLoader) {
+            unusedProvidersByType.values()
+                    .forEach(it -> result.add(it.create(Config.empty(), it.configKey())));
+        }
+
+        return result;
+    }
+
+    private static ConfiguredService configuredService(Config serviceConfig, boolean isList) {
+        if (isList) {
+            // order is significant
+            String type = serviceConfig.get("type").asString().orElse(null);
+            String name = type;
+            boolean enabled = serviceConfig.get("enabled").asBoolean().orElse(true);
+
+            Config usedConfig = serviceConfig;
+            if (type == null) {
+                // nested approach (we are on the first node of the list, we need to go deeper)
+                List<Config> configs = serviceConfig.asNodeList().orElseGet(List::of);
+                if (configs.size() != 1) {
+                    throw new ConfigException(
+                            "Service provider configuration defined as a list must have a single node that is the type, "
+                                    + "with children containing the provider configuration. Failed on: " + serviceConfig.key());
+                }
+                usedConfig = configs.get(0);
+                name = usedConfig.name();
+                type = usedConfig.get("type").asString().orElse(name);
```

Comment:
i think these "special" attributes should be teased out into a well defined interface, and add const string key values to describe them and refer to those const strings here.  `type`, `enabled`, probably/optionally `name` too...

Reply:
Done

Reply:
Can you summarize the changes - I don't see the updates/commits.

Reply:
`ProvidedUtil#KEY_SERVICE_TYPE`, and other constants on the class